CMakeLists.txt

```diff
@@ -1,3 +1,32 @@
+if(NOT CMAKE_SOURCE_DIR STREQUAL CMAKE_CURRENT_SOURCE_DIR)
+
+# Generated by `boostdep --cmake hana`
+# Copyright 2020, 2021 Peter Dimov
+# Distributed under the Boost Software License, Version 1.0.
+# https://www.boost.org/LICENSE_1_0.txt
+
+cmake_minimum_required(VERSION 3.8...3.20)
+
+project(boost_hana VERSION "${BOOST_SUPERPROJECT_VERSION}" LANGUAGES CXX)
+
+add_library(boost_hana INTERFACE)
+add_library(Boost::hana ALIAS boost_hana)
+
+target_include_directories(boost_hana INTERFACE include)
+
+target_link_libraries(boost_hana
+  INTERFACE
+    Boost::config
+    Boost::core
+    Boost::fusion
+    Boost::mpl
+    Boost::tuple
+)
```

Comment:
So those are made available by means of a superproject including this file and defining those targets?

Reply:
Usually, yes. Another supported workflow (or at least, a workflow I would like Boost to support) is the user manually including the required Boost dependencies with `add_subdirectory`, whether as submodules or with `FetchContent`.

All Boost libraries are expected to define a `Boost::libname` target.

(This should also work with a CMake-installed Boost - the installed config file for a library defines the `Boost::libname` imported target.)